### ISSUE TITLE
Incompatible with modded accessories

### ISSUE THREAD

**Issue by peow** [Exopandora/ShoulderSurfing#174]
### Description

When the mod makes you more transparent, some accessories added by other mods are not affected, in this screenshot I am using a crown from Clutter mod, the backpack from the Inmis and InmisAddon mod, and weapons attached to the body from the YDM's Weapon Master mod

![2024-04-26_21 53 31](https://github.com/Exopandora/ShoulderSurfing/assets/105888751/56e7ac1d-c43a-4ef8-aaf4-9998d524c671)

https://modrinth.com/mod/clutter
https://modrinth.com/mod/inmis
https://modrinth.com/mod/inmisaddon
https://modrinth.com/mod/weaponmaster

--------------

If it is possible to add compatibility, I ask you to publish an update for 1.20.1

### Minecraft version

1.20.1

### Incompatible mod name

**Clutter** 1.20.(0-1)-0.5.7 / **YDM's Weapon Master** 3.0.5+1.20.1 / **Inmis** 2.7.2+1.20.1 / **InmisAddon** 1.0.4+1.20.1

### Incompatible mod version

.

### Game Logs

https://mclo.gs/AT2STET

**Comment by Exopandora** [comment]
Thank you for the report. I looked into this and here are my findings:
* Clutter renders hats as items and not as armor (see why this creates a problem below)
* Inmis renderes the backpack as an item (see why this creates a problem below)
* InmisAddon renders the backpack in a way, such that no transparency can be applied to it (see source code [here](https://github.com/Globox1997/InmisAddon/blob/1.20/src/main/java/net/inmisaddon/model/BackpackModel.java#L23))
* Weaponmaster is a closed source mod, but I suspect that it also renders its visuals as items  (see why this creates a problem below)

The problem with items:  
Due to the way minecraft is programmed, it is not possible to implement item/block model transparency on item models, only while rendering the player. This alone would already require a lot of changes to the vanilla minecraft codebase, including a lot of hardcoded transparency rules for individual items. Furthermore, the required changes might introduce visual bugs elsewhere in the game, which can already be observed when looking at water through a glass block and a nether portal . There is also no guarantee that the changes fix the compatibility with every other mod.

To summarize, it is unfortunately not possible to implement transparency for item models - at least currently. As a workaround, I suggest disabling the transparency feature alltogether in the config, if you dislike the visual bug.

**Comment by Exopandora** [comment]
I opened a PR on their github with the required changes.
